packages/core/admin/admin/src/content-manager/hooks/useRelation/useRelation.js

```diff
@@ -1,13 +1,20 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { useInfiniteQuery } from 'react-query';
 
 import { axiosInstance } from '../../../core/utils';
 
-export const useRelation = (cacheKey, { relation, search }) => {
-  const [searchParams, setSearchParams] = useState({});
+import { normalizeRelations } from '../../components/RelationInputDataManager/utils';
 
+export const useRelation = (cacheKey, { name, relation, search }) => {
+  const [searchParams, setSearchParams] = useState({});
+  const [currentPage, setCurrentPage] = useState(0);
+  /**
+   * This runs in `useInfiniteQuery` to actually fetch the data
+   */
   const fetchRelations = async ({ pageParam = 1 }) => {
     try {
+      setCurrentPage(pageParam);
```

Comment:
Should this be set BEFORE the call or after? In case an error occurs I think we do not want to set the page ...

Reply:
yep good point.